tests/qchem/test_structure.py

```diff
@@ -193,22 +193,22 @@ def test_excitations_to_wires_exceptions(singles, doubles, wires, message_match)
             np.array([3.90575761, -1.89772083, -1.36689032]),
             [[0, 2], [1, 3]],
             [[0, 1, 2, 3]],
-            [-0.14619406, -0.06502792, 0.14619406, 0.06502792],
+            [0.14619406, 0.06502792, -0.14619406, -0.06502792],  # computed with AllSinglesDoubles
```

Comment:
Should the evolution of the state with the given gates be equivalent too for both `UCCSD` and `AllSinglesDoubles`?

When I run the following:
```
weights = np.array([3.90575761, -1.89772083, -1.36689032])
singles, doubles = [[0, 2], [1, 3]], [[0, 1, 2, 3]]
s_wires, d_wires = qml.qchem.excitations_to_wires(singles, doubles)
n = 4
wires = range(n)

dev = qml.device("default.qubit", wires=n)

@qml.qnode(dev)
def scircuit1(weights):
    # flipped because executed on Cloud
    qml.UCCSD(weights, wires=wires, s_wires=s_wires, d_wires=d_wires, 
                          init_state=np.array([0, 0, 1, 1]))
    return qml.state()

@qml.qnode(dev)
def scircuit2(weights):
    qml.AllSinglesDoubles(weights, wires=wires, singles=singles, doubles=doubles, 
                                            hf_state=np.array([1, 1, 0, 0]))
    return qml.state()

np.round(scircuit1(weights), 8), scircuit2(weights)
```
I get these: 
```
(tensor([ 0.        +0.j,  0.        +0.j,  0.        +0.j,
         -0.7219378 +0.j,  0.        -0.j, -0.        +0.j,
         -0.22782196+0.j,  0.        -0.j,  0.        -0.j,
         -0.10639443+0.j,  0.        +0.j,  0.        -0.j,
         -0.64465742+0.j,  0.        +0.j,  0.        -0.j,
         -0.        +0.j], requires_grad=True),
 tensor([ 0.        +0.j,  0.        +0.j,  0.        +0.j,
         -0.7219378 +0.j,  0.        +0.j,  0.        +0.j,
         -0.22782196+0.j,  0.        +0.j,  0.        +0.j,
          0.10639443+0.j,  0.        +0.j,  0.        +0.j,
         -0.64465742+0.j,  0.        +0.j,  0.        +0.j,
          0.        +0.j], requires_grad=True))
```
I see an extra negative sign on `0.10639443` (the coeff of |1001>) in the UCCSD case.

Reply:
After looking into this further, I realized that `qml.SingleExcitation` and `qml.FermionicSingleExcitation` (and similarly the `DoubleExcitation`) are not always equivalent and are related by the following relation [[1](https://arxiv.org/pdf/2005.14475.pdf)], where `P` denotes parity of the qubits:
<img width="606" alt="image" src="https://user-images.githubusercontent.com/23234279/194750076-47d5e1b0-acc2-4257-a87c-d4878469ff15.png">
Therefore, `AllSinglesDoubles` template will not always be equivalent to the `UCCSD` template. So I think it would be better to retain the previous values.

Reply:
> So I think it would be better to retain the previous values.
The previous values are for the flipped state [0, 0, 1, 1]. Removed the reference to AllSinglesDoubles.